include/mqtt/property.hpp

```diff
@@ -700,11 +569,30 @@ class user_property {
     }
 
     static constexpr detail::ostream_format const of_ = detail::ostream_format::key_val;
+
+private:
+    struct len_str {
+        explicit len_str(buffer b, bool already_checked = false)
+            : buf(std::move(b)),
+              len{ num_to_2bytes(static_cast<std::uint16_t>(buf.size())) }
```

Comment:
boost::numeric_cast?

That way, if the current value of the buffer.size() is larger than max uint16_t, an exception is thrown.

Reply:
Fixed.